src/src/pages/capabilities/jsonmetadata/index.js

```diff
@@ -0,0 +1,203 @@
+import React, { useContext, useEffect, useState } from "react";
+import SelectedCapabilityContext from "../SelectedCapabilityContext";
+import AppContext from "../../../AppContext";
+import { Button, ButtonStack, Spinner } from "@dfds-ui/react-components";
+import Ajv from "ajv";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import styles from "./jsonmetadata.module.css";
+import PageSection from "../../../components/PageSection";
+import MonacoEditor, { useMonaco } from "@monaco-editor/react";
+import { vs as syntaxStyle } from "react-syntax-highlighter/dist/esm/styles/hljs";
+export function JsonMetadataWithSchemaViewer() {
+  const monaco = useMonaco();
+  const { metadata, setCapabilityJsonMetadata } = useContext(
+    SelectedCapabilityContext,
+  );
+  const [isEditing, setIsEditing] = useState(false);
+  const [isDirty, setIsDirty] = useState(false);
+  const { selfServiceApiClient } = useContext(AppContext);
+  const [currentMetadataString, setCurrentMetadataString] = useState(metadata);
+  const [jsonString, setJsonString] = useState("");
+  const [schemaString, setSchemaString] = useState("");
+  const [validationError, setValidationError] = useState("");
+
+  useEffect(() => {
+    if (monaco && schemaString) {
+      const monacoSchema = {
+        fileMatch: ["*"],
+        schema: JSON.parse(schemaString),
+      };
+      monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+        validate: true,
+        schemas: [monacoSchema],
+      });
+    }
+  }, [schemaString, monaco]);
+
+  useEffect(() => {
+    if (metadata) {
+      const prettyMetadata = prettifyJsonString(metadata);
+      setCurrentMetadataString(prettyMetadata);
+      setJsonString(prettyMetadata);
+    }
+  }, [metadata]);
+
+  useEffect(() => {
+    if (schemaString === "") {
+      selfServiceApiClient.getCapabilityJsonMetadataSchema().then((schema) => {
+        setSchemaString(prettifyJsonString(schema));
+      });
+    }
+  }, []);
+
+  const checkIfJsonIsParsable = (json) => {
+    try {
+      JSON.parse(json);
+      return true;
+    } catch (exception) {
+      setValidationError("Invalid JSON: " + exception.message);
+      return false;
+    }
+  };
+
+  const checkIfFollowsSchema = (json) => {
```

Comment:
```suggestion
  const checkIfJsonFollowsSchema = (json) => {
```

This aligns with the previous method

Reply:
renamed

Reply:
This is still not aligned with the previous method as far as I can see.
Not a big issue, so 🤷